File: scenario3_unfixable/parser.py
"""A deliberately ambiguous module - the test expects impossible behavior."""


def parse_config(raw: str) -> dict:
    """Parses a simple key=value config string."""
    result = {}
    for line in raw.strip().split("\n"):
        if "=" in line:
            key, value = line.split("=", 1)
            result[key.strip()] = value.strip()
    return result

File: scenario3_unfixable/Parser.kt
fun parseConfig(raw: String): Map<String, String> {
    val result = mutableMapOf<String, String>()
    raw.trim().split("\n").forEach { line ->
        if (line.contains("=")) {
            val (key, value) = line.split("=", limit = 2)
            result[key.trim()] = value.trim()
        }
    }
    return result
}
